### PR TITLE
task #6 - sql database scripts

### LINKED ISSUE #6: Creating script for potential game storage SQL database
Writing a SQL script for the creation of a database with some useful tables for a Dao game db. There is potential for the implementation of a SQL database with the game server, and this script can be referred to to quickly create the necessary objects.

Also using this task to test out the functionality of GitHub Pull Requests

### PR DESCRIPTION
**Pull request by milkshake-moo**

### CODE REVIEW

File: Database/database_scripts.sql [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    player_id int PRIMARY KEY,
+    player_name varchar(25)
+);
+        INSERT INTO [plyr].[players] (player_id, player_name)
+        VALUES ( (1, "Blapo"), (2,"donny"), (69, "zaser") );
+
+/* Create tables for other game statistics */
+USE DAOdb;
+GO
+CREATE TABLE [fun].[fun_positions] (
```
Comment (donny138) [inline Database/database_scripts.sql:62]: I love the table name XD

File: Database/database_scripts.sql [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/* This file can be run to create the necessary objects for
+    a Dao game database */
+
+/* Create the Database */
+CREATE DATABASE DAOdb;
+GO
```
Comment (donny138) [inline Database/database_scripts.sql:6]: GO!  lol I didn't realize this was how you told the database to start

File: Database/database_scripts.sql [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+/* Create the Database */
+CREATE DATABASE DAOdb;
+GO
+
+/* Create database objects schemas */
+USE DAOdb;
+GO
+CREATE SCHEMA game
+CREATE SCHEMA plyr
+CREATE SCHEMA fun
```
Comment (donny138) [inline Database/database_scripts.sql:13]: Are creating these schemas like declaring a variable?  Or is it like creating a name that you can reference and ties stuff together?

File: Database/database_scripts.sql [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+/* Create dimensions for player data */
+USE DAOdb;
+GO
+CREATE TABLE [plyr].[players] (
+    player_id int PRIMARY KEY,
+    player_name varchar(25)
+);
+        INSERT INTO [plyr].[players] (player_id, player_name)
+        VALUES ( (1, "Blapo"), (2,"donny"), (69, "zaser") );
```
Comment (donny138) [inline Database/database_scripts.sql:57]: These are fantastic initial table values XD

**Review by donny138 — APPROVED** [review]
This looks cool!  I don't know pretty much any sql, but I think I'm following what this does!

Your changes are officially approved sir

File: Database/database_scripts.sql (hunk 1 above)
Comment (milkshake-moo) [reply Database/database_scripts.sql:62]: :)

File: Database/database_scripts.sql (hunk 2 above)
Comment (milkshake-moo) [reply Database/database_scripts.sql:6]: "GO" is fun to use. It's not even really necessary most of the time but I like using it when working directly with the structures!

File: Database/database_scripts.sql (hunk 3 above)
Comment (milkshake-moo) [reply Database/database_scripts.sql:13]: Schemas help you organize a database. For our DAO db, they aren't really necessary, but for larger databases it allows you to organize and probably most importantly can be set up with permissions. 
I set up DAO db with these three to start, 

It'll look like this:
DAOdb:
       fun.fun_positions
       ...
       game.history_log
       game.game_results
       ...
       plyr.players
       ...
       
So when you reference a table or a function or whatever, the full path will include the schema. If I wanted to, I could assign different login credentials to different user rights groups and allow access to each schema separately.

File: Database/database_scripts.sql (hunk 4 above)
Comment (milkshake-moo) [reply Database/database_scripts.sql:57]: :)

**Pull request merged**